### DISCUSSION TITLE
Smart home panel and delta pro

### DISCUSSION THREAD

**Discussion by Eric8383** [tolwi/hassio-ecoflow-cloud#212] [Ideas]
Hello,

How to automate charge control/choice with a Smart home panel.
Configuration is 2 delta pro with a Smart home panel.

Need another api ? 

Please help 

Eric

**Comment by TeStevens315** [comment]
Plus one for smart home panel.

**Comment by Maxctran** [comment]
I am not sure if these discussions are being read @tolwi but please can you add Smart Home Panel to be supported it will help complete the automation I want thanks

**Comment by mauolaldekrs** [comment]
Interested on collaborate to add the smart home panel with delta pro to this integration, anyone could help ?

**Comment by KliGG** [comment]
Hi,

is it planed to integrate the first version of the smart home panel?

**Comment by emdimon** [comment]
Heads-up for everyone watching this thread since Feb 2024: I went ahead and reverse-engineered the SHP charging-schedule write path over EcoFlow's private MQTT broker and published a small Python library:

**🔗 https://github.com/emdimon/ecoflow-shp-mqtt**

Tested against EU region (`api-e.ecoflow.com` / `mqtt-e.ecoflow.com:8883`) with one **SHP gen 1 (SP10)** + two Delta Pros, firmware as of May 2026.

The schedule write turns out to be plain JSON (no protobuf needed) on `cmdSet:11, id:81` published to `/app/{user_id}/{shp_serial}/thing/property/set`. It's the same call the EcoFlow mobile app makes when you edit the schedule on your phone. The library covers:

- `chChargeWatt` — per-DP AC charging rate in W
- `hightBattery` — charge cap SOC % (the typo `hight` is in EcoFlow's actual protocol)
- `lowBattery` — discharge floor SOC %
- `timeScale` — 18-byte / 144-bit / 10-minute-resolution time-of-day bitmask
- `set_reply` verification with retry

Three example scripts (`set_schedule.py`, `get_schedule.py`, `capture.py`) plus a discovery doc and full payload schema reference.

I'm not planning a drive-by PR here — the library is small (~370 lines) and MIT-licensed, so anyone who wants to wire it into HA entities can do so. Happy to help review if someone takes that on, and especially interested in capture logs from SHP 2 / SHP 3 / Ocean owners since I only have a gen-1 SP10.
